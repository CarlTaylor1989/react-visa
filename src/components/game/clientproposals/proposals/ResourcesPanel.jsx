import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { I18n, Translate } from 'react-redux-i18n';
import ScrollBar from 'react-scrollbars-custom';
import ResourceButton from './ResourceButton';
import ResourcePopup from './resources/ResourcePopup';
import scrollBarSettings from '../../../../config/scrollBar';

import './ResourcesPanel.scss';

class ResourcesPanel extends Component {
  constructor(props) {
    super(props);
    this.clientResources = this.getClientResources();
    this.getClientResources = this.getClientResources.bind(this);
  }

  getClientResources() {
    const { client } = this.props;
    return I18n.t(`clientprops.clients.${client}.resources`);
  }

  render() {
    const {
      client,
      isResourcePopupOpen,
      selectedResourceType,
      selectedResourceId,
      onOpenResourcePopup,
      onCloseResourcePopup
    } = this.props;

    this.resources = Object.keys(this.clientResources).map(resourceId => (
      <ResourceButton
        key={resourceId}
        id={resourceId}
        client={client}
        handleClick={onOpenResourcePopup}
      />
    ));

    return (
      <div className="resourcesPanel">
        <Translate className="title" value="clientprops.resourcesTitle" />
        <Translate className="text" value="clientprops.resourcesText" />
        <div className="scrollingArea">
          <ScrollBar {...scrollBarSettings}>
            <div className="resourcesWrapper">{this.resources}</div>
          </ScrollBar>
        </div>
        <ResourcePopup
          showPopup={isResourcePopupOpen}
          closePopup={onCloseResourcePopup}
          type={selectedResourceType}
          resourceId={selectedResourceId}
          client={client}
          resources={this.clientResources}
        />
      </div>
    );
  }
}

ResourcesPanel.propTypes = {
  client: PropTypes.string.isRequired,
  isResourcePopupOpen: PropTypes.bool.isRequired,
  selectedResourceType: PropTypes.string.isRequired,
  selectedResourceId: PropTypes.string.isRequired,
  onOpenResourcePopup: PropTypes.func.isRequired,
  onCloseResourcePopup: PropTypes.func.isRequired,
};

export default ResourcesPanel;
