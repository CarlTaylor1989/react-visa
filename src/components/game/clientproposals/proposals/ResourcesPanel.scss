@import '../../../../styles/variables';
@import '../../../../styles/generic';

$border-color: rgba(255, 255, 255, .3);

.resourcesPanel {
  @include multi-angle-gradient(
    180deg,
    rgba($colour-blue-dark, .4) 0%,
    rgba($colour-blue-dark, 0) 6.9%,
    rgba($colour-blue-dark, 0) 94.9%,
    rgba($colour-blue-dark, .4) 100%,
  );
  @include border-radius(27px);
  border: 2px solid $border-color;
  box-sizing: border-box;
  height: 570px;
  margin-top: 100px;
  padding: 15px 25px 20px;
  position: relative;
  width: 450px;

  .title {
    color: $colour-gold;
    display: block;
    font-size: 30px;
    margin-bottom: 20px;
  }

  .text {
    display: block;
    font-size: 18px;
    padding: 0 60px 15px;
    position: relative;

    &::after {
      background-color: $colour-white;
      bottom: 0;
      content: '';
      display: block;
      height: 1px;
      left: 50%;
      position: absolute;
      transform: translate(-50%, 0);
      width: 230px;
    }
  }

  > .scrollingArea {
    @extend %scrollbar;
    height: 385px;
    margin-top: 20px;

    .resourcesWrapper {
      display: flex;
      flex-wrap: wrap;
      padding: 0 30px;
    }
  }

  .resourcePopupOverlay {
    @include border-radius(27px);
    border-radius: 27px;
    position: absolute;
  }
}
