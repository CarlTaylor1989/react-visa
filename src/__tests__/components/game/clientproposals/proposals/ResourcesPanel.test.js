/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import { I18n } from 'react-redux-i18n';
import ResourcesPanel from
  '../../../../../components/game/clientproposals/proposals/ResourcesPanel';

I18n.t = jest.fn((location) => {
  if (location.match(/resources$/)) {
    return {
      0: {
        id: 'res1',
        name: 'RESOURCE NAME',
        text: 'TEXT',
        type: 'pdf',
        url: 'url'
      },
      1: {
        id: 'res2',
        name: 'RESOURCE NAME',
        text: 'TEXT',
        type: 'video',
        url: 'url'
      },
      2: {
        id: 'res3',
        name: 'RESOURCE NAME',
        text: 'TEXT',
        type: 'audio',
        url: 'url'
      }
    };
  }

  if (location.match(/id$/)) {
    return 'res1';
  }

  if (location.match(/type$/)) {
    return 'report';
  }

  return '';
});

let mockProps;
beforeEach(() => {
  mockProps = {
    client: 'client1',
    isResourcePopupOpen: false,
    selectedResourceType: 'type',
    selectedResourceId: 'id',
    onOpenResourcePopup: jest.fn(),
    onCloseResourcePopup: jest.fn(),
  };
});

test('renders the ResourcesPanel component', () => {
  const wrapper = shallow(<ResourcesPanel {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
