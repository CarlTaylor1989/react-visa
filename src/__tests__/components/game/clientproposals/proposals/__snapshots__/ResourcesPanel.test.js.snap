// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ResourcesPanel component 1`] = `
<div
  className="resourcesPanel"
>
  <Translate
    className="title"
    tag="span"
    value="clientprops.resourcesTitle"
  />
  <Translate
    className="text"
    tag="span"
    value="clientprops.resourcesText"
  />
  <div
    className="scrollingArea"
  >
    <Scrollbar
      contentProps={Object {}}
      fallbackScrollbarWidth={20}
      maximalThumbYSize={83}
      minimalThumbSize={30}
      minimalThumbYSize={83}
      momentum={true}
      noDefaultStyles={true}
      scrollDetectionThreshold={100}
      scrollerProps={Object {}}
      thumbXProps={Object {}}
      thumbYProps={Object {}}
      trackClickBehavior="jump"
      trackXProps={Object {}}
      trackYProps={Object {}}
      wrapperProps={Object {}}
    >
      <div
        className="resourcesWrapper"
      >
        <ResourceButton
          client="client1"
          handleClick={[MockFunction]}
          id="0"
          key="0"
        />
        <ResourceButton
          client="client1"
          handleClick={[MockFunction]}
          id="1"
          key="1"
        />
        <ResourceButton
          client="client1"
          handleClick={[MockFunction]}
          id="2"
          key="2"
        />
      </div>
    </Scrollbar>
  </div>
  <ResourcePopup
    client="client1"
    closePopup={[MockFunction]}
    resourceId="id"
    resources={
      Object {
        "0": Object {
          "id": "res1",
          "name": "RESOURCE NAME",
          "text": "TEXT",
          "type": "pdf",
          "url": "url",
        },
        "1": Object {
          "id": "res2",
          "name": "RESOURCE NAME",
          "text": "TEXT",
          "type": "video",
          "url": "url",
        },
        "2": Object {
          "id": "res3",
          "name": "RESOURCE NAME",
          "text": "TEXT",
          "type": "audio",
          "url": "url",
        },
      }
    }
    showPopup={false}
    type="type"
  />
</div>
`;
